bin/concatenate_all.py

```diff
@@ -138,145 +205,80 @@ def main():
-                                                                  'user_assignment_settings.json')
-                    if os.path.exists(other_submissions_details_path):
-                        with open(other_submissions_details_path) as other_details_file:
-                            other_details_json = json.load(other_details_file)
-                            my_active_version = int(other_details_json["active_version"])
-                    else:
-                        # get the most recent version
-                        my_active_version = sorted(os.listdir(other_user_path))[-1]
-
-                # loop over each version
-                for other_version in sorted(os.listdir(other_user_path)):
-                    other_version_path = os.path.join(other_user_path, other_version)
-                    if dir == "results":
-                        # only the "details" dir within "results" contains files relevant to Lichen
-                        other_version_path = os.path.join(other_version_path, "details")
-                    if not os.path.isdir(other_version_path):
-                        continue
-
-                    other_output_file_path = os.path.join(args.basepath, "other_gradeables",
-                                                          f"{other_gradeable['other_semester']}__{other_gradeable['other_course']}__{other_gradeable['other_gradeable']}",  # noqa: E501
-                                                          other_user, other_version,
-                                                          "submission.concatenated")
-
-                    if not os.path.exists(os.path.dirname(other_output_file_path)):
-                        os.makedirs(os.path.dirname(other_output_file_path))
-
-                    # append to concatenated file
-                    with open(other_output_file_path, "a") as other_output_file:
-                        other_concatenated_contents = getConcatFilesInDir(other_version_path,
-                                                                          regex_patterns)
-                        other_output_file.write(other_concatenated_contents)
-                        total_concat += sys.getsizeof(other_concatenated_contents)
-
-                    checkTotalSize(total_concat)
+            input_path = os.path.join(args.datapath,
+                                      other_gradeable["other_semester"],
+                                      other_gradeable["other_course"],
+                                      dir,
+                                      other_gradeable["other_gradeable"])
+
+            output_path = os.path.join(args.basepath, "other_gradeables",
+                                       f"{other_gradeable['other_semester']}__{other_gradeable['other_course']}__{other_gradeable['other_gradeable']}")  # noqa: E501
+            total_concat = processGradeable(args.basepath, config,
```

Comment:
same comment as above
```suggestion
            total_concat += processGradeable(args.basepath, config,
```